Modules/Loadable/Volumes/qSlicerVolumesReader.cxx

```diff
@@ -186,7 +191,14 @@ bool qSlicerVolumesReader::load(const IOProperties& properties)
         }
       if (appLogic)
         {
-        appLogic->PropagateVolumeSelection(); // includes FitSliceToAll by default
+        if (fit == 1)
+          {
+          appLogic->PropagateVolumeSelection(); // includes FitSliceToAll by default
+          }
+        else if (fit == 2)
+          {
+          appLogic->FitSlicesToVolume(node->GetID());
```

Comment:
If fit==1 then this call both selects the volume in the background layer and fits slices.
However, if fit==2 then only ->FitSlicesToVolume(node->GetID()) is called, which method does not selects the volume to be shown in any slice viewer.